applications/DamApplication/custom_constitutive/thermal_linear_elastic_3D_law.cpp

```diff
@@ -107,13 +97,35 @@ void ThermalLinearElastic3DLaw::CalculateMaterialResponseKirchhoff (Parameters&
         this->CalculateThermalStrain(StrainVector,ElasticVariables,Temperature);
 
         StressVector = prod(ConstitutiveMatrix,StrainVector);
-    }
-    else if(Options.Is( ConstitutiveLaw::COMPUTE_STRAIN ) ) //This should be COMPUTE_THERMAL_STRAIN
-    {
+      }
+      else{
+
+	this->CalculateLinearElasticMatrix( ConstitutiveMatrix, YoungModulus, PoissonCoefficient );
+
         double Temperature;
         this->CalculateDomainTemperature( ElasticVariables, Temperature);
+
+        Vector ThermalStrainVector;
+        this->CalculateThermalStrain(ThermalStrainVector,ElasticVariables,Temperature);
+
+        StressVector = prod(ConstitutiveMatrix,(StrainVector - ThermalStrainVector));
```

Comment:
Sorry @RiccardoRossi, now that I look carefully, it is not possible to define an array_1d<double,6> for the temporal vector because the method CalculateMaterialResponseCauchy is inside a 3D ConstitutiveLaw that is also used for 2D plane stress/plane strain laws. 

In order to avoid creating a derived method for each law, we could define a dynamic vector of Voigt size, but... would there be any improvement with respect to the current code?